### PR TITLE
feat: verify token manager proxy

### PR DESCRIPTION
**Pull request by deanamiel**
Add verify token manager proxy option to contract verification script as per [AXE-2875](https://axelarnetwork.atlassian.net/jira/software/c/projects/AXE/boards/21?selectedIssue=AXE-2875)

[AXE-2875]: https://axelarnetwork.atlassian.net/browse/AXE-2875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: evm/verify-contract.js [hunk 1]
```diff
@@ -284,6 +284,55 @@ async function processCommand(config, chain, options) {
...
+                tokenId = await interchainTokenService.interchainTokenId(deployer, deploymentSalt);
+                tokenManagerAddress = await interchainTokenService.tokenManagerAddress(tokenId);
+
+                const TokenManagerProxy = getContractJSON('TokenManagerProxy');
+                const tokenManagerProxyFactory = await getContractFactoryFromArtifact(TokenManagerProxy, wallet);
+                const tokenManagerProxy = tokenManagerProxyFactory.attach(tokenManagerAddress);
+
+                let tokenAddress;
+                [implementationType, tokenAddress] = await tokenManagerProxy.getImplementationTypeAndTokenAddress();
+                params = defaultAbiCoder.encode(['bytes', 'address'], [minter, tokenAddress]);
```
Comment (milapsheth) [inline evm/verify-contract.js:319]: pass empty 0x bytes if minter is not provided

File: evm/verify-contract.js [hunk 2]
```diff
@@ -297,14 +346,24 @@ async function main(options) {
...
+    program.addOption(new Option('--deployer <deployer>', 'interchain token deployer address'));
+    program.addOption(new Option('--salt <salt>', 'interchain token deployment salt'));
+    program.addOption(new Option('--rawSalt <rawSalt>', 'raw deployment salt').env('RAW_SALT'));
+    program.addOption(new Option('--minter <minter>', 'interchain token minter address'));
+    program.addOption(new Option('--tokenId <tokenId>', 'interchain token tokenId'));
+    program.addOption(
+        new Option('--implementationType <implementationType>', 'token manager implementation type').choices(['0', '1', '2', '3']),
+    );
+    program.addOption(new Option('--tokenManagerAddress <tokenManagerAddress>', 'token manager proxy address'));
+    program.addOption(new Option('--params <params>', 'token manager deployment params'));
```
Comment (milapsheth) [inline evm/verify-contract.js:366]: too many options, just need tokenId, and minter if it was set, and everything else should be calculated dynamically. we just need to do this once on most chains anyways

File: evm/verify-contract.js [hunk 3]
```diff
@@ -297,14 +346,24 @@ async function main(options) {
...
 
     program.action((options) => {
```
Comment (milapsheth) [inline evm/verify-contract.js:368]: add usage to readme

Comment (deanamiel) [reply evm/verify-contract.js:368]: I removed all these options now that verification will only depend on the `tokenId` and the `minter`, do you still want me to add an example to the readme? It should be just as straightforward as the other contract verification cases now

File: evm/verify-contract.js [hunk 4]
```diff
@@ -297,14 +333,19 @@ async function main(options) {
...
-    program.name('balances').description('Display balance of the wallet on specified chains.');
+    program.name('verify-contract').description('Verify selected contracts on specified chains.');
 
     addBaseOptions(program, { ignorePrivateKey: true, address: true });
 
     program.addOption(new Option('-c, --contractName <contractName>', 'contract name'));
     program.addOption(new Option('-d, --dir <dir>', 'contract artifacts dir'));
     program.addOption(new Option('--args <args>', 'contract args'));
     program.addOption(new Option('--constructorArgs <constructorArgs>', 'contract constructor args'));
+    program.addOption(new Option('--deployer <deployer>', 'interchain token deployer address'));
```
[the remark this answers is not in the document]
Comment (deanamiel) [reply evm/verify-contract.js]: removed

File: evm/verify-contract.js (hunk 3 above)
Comment (deanamiel) [reply evm/verify-contract.js:347]: Added example usage

**Pull request merged**